### ISSUE TITLE
[Table] Improve a11y support for certain browsers, selection modes

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#8915]
**monday.com sync:** #18048581311

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

@geospatialem has identified some issues with the current a11y set up of the Table, specifically in FF, and with selectionMode enabled. Additionally, Safari 17 fixed some issues with `display:contents` we had to add specific workarounds for. When possible, these should be removed.

### Acceptance Criteria

- [ ] Accommodate AT context with FF browsers for JAWS and NVDA for multiple selections
    - Header row selection reads back part of the selections/deselections (e.g., `"Selection column. 0 selected Press Enter or Space to select all rows // all 1 // 1 // de"` where Chrome reads `"Row number Selection column. all 101 selected Press Enter or Space to deselect all rows"`)
    - Individual row selections read back only the number of rows selected when selecting a new row selection (e.g., `"2"` where Chrome reads `"Row number 1 Selection column. 2 rows selected Press Enter or Space to select all rows"`)
 

 - [ ] Safari 17 fixed `display:contents` issues - we have specific Safari code to read aloud what it does not, once 17 is 2 versions old, let's remove this.

### Which Component

Table

### Priority impact

p4 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (design)

**Comment by geospatialem** [comment]
Spike to determine time estimate to address the browser and AT-related inconsistencies/reports.

**Comment by macandcheese** [comment]
Blocked by https://github.com/Esri/calcite-design-system/issues/6646
